Code/GraphMol/Descriptors/testAUTOCORR2D.cpp

```diff
@@ -0,0 +1,119 @@
+//  Created by Guillaume GODIN
+//  Copyright (C) 2012-2016 Greg Landrum
+//   @@ All Rights Reserved @@
+//
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#include <RDGeneral/Invariant.h>
+#include <GraphMol/RDKitBase.h>
+#include <GraphMol/FileParsers/MolSupplier.h>
+#include <GraphMol/FileParsers/FileParsers.h>
+#include <RDGeneral/RDLog.h>
+#include <vector>
+#include <algorithm>
+#include <fstream>
+#include <sstream>
+
+#include <GraphMol/Descriptors/AUTOCORR2D.h>
+
+void testautocorrelation() {
+  std::cout << "=>start test autocorr2D\n";
+
+  std::string pathName = getenv("RDBASE");
+
+  //std::cout << "Path: " << pathName << "\n";
+
+  std::string sdfName =
+      pathName + "/Code/GraphMol/Descriptors/test_data/PBF_egfr.sdf";
+
+  RDKit::SDMolSupplier reader(sdfName, true, false);
+
+  std::string fName = pathName+"/Code/GraphMol/Descriptors/test_data/auto2D_dragon.out";
+
+  std::ifstream instrm(fName.c_str());
+
+  std::string line;
+  std::vector<std::vector<std::string>> data;
+
+  while(std::getline(instrm, line)) {
+      std::string phrase;
+      std::vector<std::string> row;
+      std::stringstream ss(line);
+      while(std::getline(ss, phrase, '\t')) {
+          row.push_back(std::move(phrase));
```

Comment:
This is not ready to be implemented there is an issue during the computation, I don't know where